### PR TITLE
Update term-level-queries.asciidoc and normalizer.asciidoc: keyword fields with normalizer

### PR DESCRIPTION
**Pull request by wmellouli**
In **term level queries** documentation, it is not indicated that if we use a keyword field with a normalizer, term-level query will normalize terms before executing. So actually, users should perform tests to discover this behavior (it was my case).
I think, this is a very good new feature introduced in elasticsearch (before, users normalize terms manually before indexing and before searching data).

This is an example (with elasticsearch 6.2.4):

Create an index with one normalized field (use only filter lowercase):

```
PUT localhost:9200/coutries

{
  "settings": {
    "analysis": {
      "normalizer": {
        "my_normalizer": {
          "type": "custom",
          "char_filter": [],
          "filter": ["lowercase"]
        }
      }
    }
  },
  "mappings": {
    "country": {
      "properties": {
        "name": {
          "type": "keyword",
          "index": true,
          "normalizer": "my_normalizer"
        }
      }
    }
  }
}

```

Insert some documents:

```
POST localhost:9200/coutries/country/_bulk

{ "index" : { "_id" : "1" } }
{ "name" : "Netherlands" }
{ "index" : { "_id" : "2" } }
{ "name" : "Tunisia" }
{ "index" : { "_id" : "3" } }
{ "name" : "France" }
```

And this is what I get when I use term levels queries with different way (UPPERCASE, lowercase, mixed):

- **Terms query**

```
POST localhost:9200/coutries/country/_search

{
    "query": {
        "terms" : { "name" : ["netherlands", "TUNISIA", "France"]}
    }
}

=> Response
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 3,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "2",
        "_score": 1,
        "_source": {
          "name": "Tunisia"
        }
      },
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "1",
        "_score": 1,
        "_source": {
          "name": "Netherlands"
        }
      },
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "3",
        "_score": 1,
        "_source": {
          "name": "France"
        }
      }
    ]
  }
}

```

So as we can see here,  term level queries will normalize terms string before executing for keyword field.

### CODE REVIEW

File: docs/reference/query-dsl/term-level-queries.asciidoc [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 While the <<full-text-queries,full text queries>> will analyze the query
 string before executing, the _term-level queries_ operate on the exact terms
-that are stored in the inverted index.
+that are stored in the inverted index, except for keyword fields with <<normalizer,normalizer>> property,
```
Comment (cbuescher) [inline docs/reference/query-dsl/term-level-queries.asciidoc]: "except" here might sound like term level queries operate on something other than terms that are stored in the inverted index, while this is actually the case. The input for keyword fields will just be normalized before storing it, and so will the query. 
I understand the desire to clarify this but I am on the fence of doing it here, since the purpose of this paragraph is mostly to list all the term queries, not to go into details about the analysis/normalization process. I'm wondering if there is another place in the docs where this might fit better. Alternatively, could you rephrase it so it reflects what I mentioned before, but in a very brief way so it doesn't distract much from the main purpose of this section?

**Review by cbuescher — CHANGES_REQUESTED** [review]
Hi @wmellouli, thanks for the suggestion. I understand the need to clear this up, however I left a comment explaining why I think it might be good to move this elsewhere or at least rephrase it a bit. Let me know what you think.

File: docs/reference/query-dsl/term-level-queries.asciidoc (hunk 1 above)
Comment (wmellouli) [reply docs/reference/query-dsl/term-level-queries.asciidoc]: Thank you @cbuescher for your review :+1: 
I agree that this page is mostly to list only all the term queries. But the first line is discussing about **full text queries that analyze query string before executing**, and that term level queries operate on **the exact terms**: I think this is  wrong, because term level queries will normalize terms string before executing (in keyword field case) and will not operate on the exact terms.

This is an example (with elasticsearch 6.2.4):

Create an index with one normalized field (use only filter lowercase):

```
PUT localhost:9200/coutries

{
  "settings": {
    "analysis": {
      "normalizer": {
        "my_normalizer": {
          "type": "custom",
          "char_filter": [],
          "filter": ["lowercase"]
        }
      }
    }
  },
  "mappings": {
    "country": {
      "properties": {
        "name": {
          "type": "keyword",
          "index": true,
          "normalizer": "my_normalizer"
        }
      }
    }
  }
}

```

Insert some documents:

```
POST localhost:9200/coutries/country/_bulk

{ "index" : { "_id" : "1" } }
{ "name" : "Netherlands" }
{ "index" : { "_id" : "2" } }
{ "name" : "Tunisia" }
{ "index" : { "_id" : "3" } }
{ "name" : "France" }
```

And this is what I get when I use term levels queries with different way (UPPERCASE, lowercase, mixed):

- **Term query**

```
POST localhost:9200/coutries/country/_search

{
  "query": {
    "term" : { "name" : "NETHERLANDS" } 
  }
}

=> Response
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 0.2876821,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "1",
        "_score": 0.2876821,
        "_source": {
          "name": "Netherlands"
        }
      }
    ]
  }
}
```

- **Terms query**

```
POST localhost:9200/coutries/country/_search

{
    "query": {
        "terms" : { "name" : ["netherlands", "TUNISIA", "France"]}
    }
}

=> Response
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 3,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "2",
        "_score": 1,
        "_source": {
          "name": "Tunisia"
        }
      },
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "1",
        "_score": 1,
        "_source": {
          "name": "Netherlands"
        }
      },
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "3",
        "_score": 1,
        "_source": {
          "name": "France"
        }
      }
    ]
  }
}

```

- **Prefix query**

```
POST localhost:9200/coutries/country/_search

{ "query": {
    "prefix" : { "name" : "tun" }
  }
}

=> Response
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "2",
        "_score": 1,
        "_source": {
          "name": "Tunisia"
        }
      }
    ]
  }
}
```

- **Wildcard query**

```
POST localhost:9200/coutries/country/_search

{
    "query": {
        "wildcard" : { "name" : "f*CE" }
    }
}

=> Response
{
  "took": 5,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 1,
    "hits": [
      {
        "_index": "coutries",
        "_type": "country",
        "_id": "3",
        "_score": 1,
        "_source": {
          "name": "France"
        }
      }
    ]
  }
}
```

So as we can see here,  term level queries will not operate on the exact terms but will normalize terms string before executing for keyword field.

What I suggest is:

- rephrase with this way (maybe you have a better suggestion): 

```
While the <<full-text-queries,full text queries>> will analyze the query string before executing, 
the _term-level queries_ operate on the exact terms that are stored in the inverted index, and 
will normalize terms for keyword fields with <<normalizer,normalizer>> property.
```

- modify normalizer documentation page too: https://www.elastic.co/guide/en/elasticsearch/reference/master/normalizer.html

Instead of:

`The normalizer is applied prior to indexing the keyword, as well as at search-time when the keyword field is searched via a query parser such as the match query.`

We can write:

`The normalizer is applied prior to indexing the keyword, as well as at search-time.`

Or 

`The normalizer is applied prior to indexing the keyword, as well as at search-time when the keyword field is searched via a query parser such as the match query or via a term level queries such as the term query.`


What do you think @cbuescher ? (I will rephrase, format and submit changes after this messages exchange)

**Comment by wmellouli** [comment]
@cbuescher I pushed a second commit to rephrase and modify normalizer documentation. I left a comment explaining this commit. Let me know if this commit does match or not  what you expect.

**Review by cbuescher — COMMENTED** [review]
Looks great, thanks for the changes. I left two very small requests but other than that I'm happy to merge now.

File: docs/reference/mapping/params/normalizer.asciidoc [hunk 2]
```diff
@@ -7,7 +7,8 @@ produces a single token.
 
 The `normalizer` is applied prior to indexing the keyword, as well as at
 search-time when the `keyword` field is searched via a query parser such as
-the <<query-dsl-match-query,`match`>> query.
+the <<query-dsl-match-query,`match`>> query or via a term level queries
```
[the remark this answers is not in the document]
Comment (wmellouli) [reply docs/reference/mapping/params/normalizer.asciidoc]: fixed: use query instead of queries

File: docs/reference/query-dsl/term-level-queries.asciidoc [hunk 3]
```diff
@@ -3,7 +3,8 @@
 
 While the <<full-text-queries,full text queries>> will analyze the query
 string before executing, the _term-level queries_ operate on the exact terms
-that are stored in the inverted index.
+that are stored in the inverted index, and will normalize terms before executing 
+only for <<keyword,keyword>> fields with <<normalizer,normalizer>> property.
```
[the remark this answers is not in the document]
Comment (wmellouli) [reply docs/reference/query-dsl/term-level-queries.asciidoc]: fixed: added backticks arround normalizer and keyword

**Comment by wmellouli** [comment]
Thanks for you quick response. I fixed with what you suggested.

**Review by cbuescher — APPROVED** [review]
LGTM, thanks for the quick feedback. I will run our tests once and then merge if everything looks fine.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by wmellouli** [comment]
@cbuescher elasticsearch-ci build fail with:

> 18:05:46 Execution failed for task ':distribution:bwc:next-minor-snapshot:buildBwcVersion'.
> 18:05:46 > Building bwc version didn't generate expected files [/var/lib/jen...

Any idea ?

**Comment by cbuescher** [comment]
@wmellouli we probably increased the version numbers due to a minor release in the meantime. Could you try to merge in master? I will kick of another CI run afterwards. Thanks.

**Comment by wmellouli** [comment]
@cbuescher master was merged. You can kick another CI :wink:. Thanks

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Comment by wmellouli** [comment]
@cbuescher CI is OK. You can squash & merge. Thanks
